File: packages/java/src/main/java/com/tesote/sdk/models/Account.java
package com.tesote.sdk.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Account record (v1 and v2 share the same shape).
 *
 * <p>Balance fields inside {@link AccountData} only appear when the
 * workspace has {@code display_balances_in_api} enabled.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record Account(
        @JsonProperty("id") String id,
        @JsonProperty("name") String name,
        @JsonProperty("data") AccountData data,
        @JsonProperty("bank") Bank bank,
        @JsonProperty("legal_entity") LegalEntity legalEntity,
        @JsonProperty("tesote_created_at") String tesoteCreatedAt,
        @JsonProperty("tesote_updated_at") String tesoteUpdatedAt
) {
    @JsonIgnoreProperties(ignoreUnknown = true)
    public record AccountData(
            @JsonProperty("masked_account_number") String maskedAccountNumber,
            @JsonProperty("currency") String currency,
            @JsonProperty("transactions_data_current_as_of") String transactionsDataCurrentAsOf,
            @JsonProperty("balance_data_current_as_of") String balanceDataCurrentAsOf,
            @JsonProperty("custom_user_provided_identifier") String customUserProvidedIdentifier,
            @JsonProperty("balance_cents") String balanceCents,
            @JsonProperty("available_balance_cents") String availableBalanceCents
    ) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record Bank(@JsonProperty("name") String name) {}

    @JsonIgnoreProperties(ignoreUnknown = true)
    public record LegalEntity(
            @JsonProperty("id") String id,
            @JsonProperty("legal_name") String legalName
    ) {}
}

File: packages/php/src/Models/Account.php
<?php

declare(strict_types=1);

namespace Tesote\Sdk\Models;

/** Account model (v1 + v2 — identical wire shape). */
final class Account
{
    public function __construct(
        public readonly string $id,
        public readonly string $name,
        public readonly AccountData $data,
        public readonly Bank $bank,
        public readonly ?LegalEntity $legalEntity,
        public readonly string $tesoteCreatedAt,
        public readonly string $tesoteUpdatedAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        $rawData = is_array($data['data'] ?? null) ? $data['data'] : [];
        $rawBank = is_array($data['bank'] ?? null) ? $data['bank'] : [];
        $rawLegal = is_array($data['legal_entity'] ?? null) ? $data['legal_entity'] : null;

        return new self(
            id: (string) ($data['id'] ?? ''),
            name: (string) ($data['name'] ?? ''),
            data: AccountData::fromArray($rawData),
            bank: Bank::fromArray($rawBank),
            legalEntity: $rawLegal !== null ? LegalEntity::fromArray($rawLegal) : null,
            tesoteCreatedAt: (string) ($data['tesote_created_at'] ?? ''),
            tesoteUpdatedAt: (string) ($data['tesote_updated_at'] ?? ''),
        );
    }
}
